core/src/tx.rs

```diff
@@ -45,6 +45,34 @@ pub enum AcceptTransactionFail {
     SignatureVerification(#[source] SignatureVerificationFail<TransactionPayload>),
 }
 
+fn instruction_size(isi: &InstructionBox) -> usize {
+    use InstructionBox::*;
+
+    match isi {
+        Register(isi) => isi.object.len() + 1,
+        Unregister(isi) => isi.object_id.len() + 1,
+        Mint(isi) => isi.destination_id.len() + isi.object.len() + 1,
+        Burn(isi) => isi.destination_id.len() + isi.object.len() + 1,
+        Transfer(isi) => isi.destination_id.len() + isi.object.len() + isi.source_id.len() + 1,
+        If(isi) => {
+            let otherwise = isi.otherwise.as_ref().map_or(0, instruction_size);
+            isi.condition.len() + instruction_size(&isi.then) + otherwise + 1
+        }
+        Pair(isi) => {
+            instruction_size(&isi.left_instruction) + instruction_size(&isi.right_instruction) + 1
+        }
+        Sequence(isi) => isi.instructions.iter().map(instruction_size).sum::<usize>() + 1,
+        SetKeyValue(isi) => isi.object_id.len() + isi.key.len() + isi.value.len() + 1,
+        RemoveKeyValue(isi) => isi.object_id.len() + isi.key.len() + 1,
+        Grant(isi) => isi.object.len() + isi.destination_id.len() + 1,
+        Revoke(isi) => isi.object.len() + isi.destination_id.len() + 1,
+        SetParameter(isi) => isi.parameter.len() + 1,
+        NewParameter(isi) => isi.parameter.len() + 1,
+        Upgrade(isi) => isi.object.len() + 1,
+        Fail(_) | ExecuteTrigger(_) => 1,
+    }
```

Comment:
1. What does magic $+ 1$ means? Small comment or self-descriptive const here might help to understand it.
2. I see that $+ 1$ is applied for each instruction. Probably, it would be cleaner:
   ```rs
   let base_size = match isi { /* ... */ };
   base_size + 1
   ```

Reply:
opened an issue #3726